sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Scenario.Tests/tests/EventGridBlobTriggerEndToEndTests.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.Storage.Blobs.Triggers;
+using Microsoft.Azure.WebJobs.Extensions.Storage.Common.Tests;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using NUnit.Framework;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.ScenarioTests
+{
+    public class EventGridBlobTriggerEndToEndTests : LiveTestBase<WebJobsTestEnvironment>
+    {
+        private const string TestArtifactPrefix = "e2etests";
+        private const string EventGridContainerName = TestArtifactPrefix + "eventgrid-%rnd%";
+        private const string TestBlobName = "test";
+        private readonly string _resolvedContainerName;
+
+        private readonly BlobContainerClient _testContainer;
+        private readonly BlobServiceClient _blobServiceClient;
+        private readonly RandomNameResolver _nameResolver;
+
+        private const string RegistrationRequest =
```

Comment:
Is it OK to use hardcoded requests ?
These tests will run in https://dev.azure.com/azure-sdk/internal/_build?definitionId=410&_a=summary with Azure resources generated on the fly via ARM.
Easiest way to check is to create branch in main repo (not fork) and run that pipeline from branch.

Reply:
There is no point in creating eventgid subscription as an webjobs extension itself does not have the http endpoint. We need function host to add the http endpoint.

So best we can do is emulating http requests(passing hardcoded payload) in the tests.

Reply:
Okay, I'm not exactly sure how this emulation works. Worst case we'll follow up if live tests fail.